tests/sample-tester.rs

```diff
@@ -12,22 +12,17 @@ fn input_output_tester() {
 
     for file in input_dir {
         let file = file.unwrap();
-        let input_path = file.path();
-        let expected_path = expected_dir.join(file.file_name());
-        let extension = input_path.extension().unwrap().to_str().unwrap();
 
-        let language = match extension {
-            "json" => Language::Json,
-            "ml" => Language::Ocaml,
-            "rs" => Language::Rust,
-            "toml" => Language::Toml,
-            _ => panic!("File extension {} not supported.", extension),
-        };
+        let input_path = file.path();
+        let language =
+            Language::detect(input_path.to_str().unwrap()).unwrap_or_else(|err| panic!("{err}"));
```

Comment:
This `unwrap_or_else` could have been a simple `unwrap`, right?

Reply:
Probably :sweat_smile:

I'm probably wrong, but I thought `unwrap`'ing an arbitrary `Err` would just give a generic panic. Whereas, as in this case, when the error type implements `Display`, I can use `unwrap_or_else` to get a nicer error message... Or am I wrong and `unwrap` is also `Display`-aware?

Reply:
It looks like the `unwrap` panic uses the error type's `Debug` implementation, rather than `Display`. For example:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error', src/main.rs:14:7
```